build_dependency_tree.py

```diff
@@ -3,14 +3,27 @@
 import sys
 import re
 from pathlib import Path
+from typing import Optional
 from typing import List
+from typing import Union
 
-args = [arg for arg in sys.argv]
-files_in_dir = [
-    [file for file in Path(root).glob('*') # TODO: Make recursive search 
-     if file.suffix in ('h', 'hpp', 'c', 'cpp')]
-    for root in args
-]
+args: List[Union[bytes, str]] = [arg for arg in sys.argv]
+files_in_project = []
+
+
+def extract_files_from_directory(
+        dir_or_file: Path, list_o_files: List[Optional[Path]]
+):
+    """FUnction recurses through the directory to find all file-like objects"""
+    if dir_or_file.is_dir():
+        for path in dir_or_file.glob("*"):
+            extract_files_from_directory(path, list_o_files)
+    else:
+        list_o_files.append(dir_or_file)
+
+
+for arg in args:
```

Comment:
I don't think there should be code that executes every time imports the file. This should been wrapped in a function that the use can call or guarded by the `if __name__ == "__main__":` statement

Reply:
So what we need to do is create a main function that will call alll these utility/helper functions that we are building right now.

Reply:
Yes, though we should probably have better name than that